### PR TITLE
[ts-sdk] Transaction builder coin selection

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
## Description 

Initial version of coin selection, which currently just selects all coins. We'll refine this over time.

## Test Plan 

Tests should pass.

### CODE REVIEW

**Comment by Jordan-Mysten** [comment]
Current dependencies on/for this PR:
* main
  * **PR #8904** <a href="https://app.graphite.dev/github/pr/MystenLabs/sui/8904" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/MystenLabs/sui/8904?utm_source=stack-comment).

File: sdk/typescript/src/builder/legacy.ts [hunk 1]
```diff
@@ -123,5 +128,6 @@ export async function convertToTransactionBuilder(
     tx.setGasPrice(data.gasPrice);
   }
 
-  return tx;
+  // console.log(tx.transactionData);
```
Comment (666lcz) [inline sdk/typescript/src/builder/legacy.ts:131]: nit: remove?

**Review by 666lcz — APPROVED** [review]
Love the new API!!!

File: sdk/typescript/src/builder/legacy.ts (hunk 1 above)
Comment (Jordan-Mysten) [reply sdk/typescript/src/builder/legacy.ts:131]: Good catch, will remove

**Pull request merged**